File: examples/android/MSDKSampleAppJava/app/src/main/java/msdk/java/sample/db/dao/BackupDao.java
package msdk.java.sample.db.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import msdk.java.sample.db.entity.Backup;

@Dao
public interface BackupDao {

    @Query("SELECT * FROM backup WHERE id = :id")
    Backup getById(int id);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAll(Backup... backups);

    @Update
    void update(Backup backup);

    @Delete
    void delete(Backup backup);
}

File: examples/android/MSDKSampleAppKotlin/app/src/main/java/msdk/kotlin/sample/db/dao/BackupDao.kt
package msdk.kotlin.sample.db.dao

import androidx.room.*
import msdk.kotlin.sample.db.entity.Backup


@Dao
interface BackupDao {
    @Query("SELECT * FROM backup WHERE id = :id")
    suspend fun getById(id: Int): Backup?

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertAll(vararg backups: Backup)

    @Update
    suspend fun update(connection: Backup)

    @Delete
    suspend fun delete(connection: Backup)
}